nowcasting_forecast/models/cnn/model.py

```diff
@@ -0,0 +1,346 @@
+
+        out = out.reshape(batch_size, self.cnn_output_size)
+
+        # Fully connected layers
+        out = F.relu(self.fc1(out))
+        out = F.relu(self.fc2(out))
+        # which has shape (batch_size, 128)
+
+        # add pv yield
+        if self.include_pv_or_gsp_yield_history:
+            pv_yield_history = pv_data[:, : self.history_len_30 + 1].nan_to_num(nan=0.0).float()
+
+            pv_yield_history = pv_yield_history.reshape(
+                pv_yield_history.shape[0], pv_yield_history.shape[1] * pv_yield_history.shape[2]
+            )
+            # join up
+            out = torch.cat((out, pv_yield_history), dim=1)
+
+        # add the pv yield history. This can be used if trying to predict gsp
+        if self.include_pv_yield_history:
+            # just take the first 128
+            pv_yield_history = (
+                pv_data[:, : self.history_len_5 + 1, :128].nan_to_num(nan=0.0).float()
+            )
+
+            pv_yield_history = pv_yield_history.reshape(
+                pv_yield_history.shape[0], pv_yield_history.shape[1] * pv_yield_history.shape[2]
+            )
+            pv_yield_history = F.relu(self.pv_fc1(pv_yield_history))
+
+            out = torch.cat((out, pv_yield_history), dim=1)
+
+        # *********************** NWP Data ************************************
+        if self.include_nwp:
+
+            # shape: batch_size, n_chans, seq_len, height, width
+            # nwp_data = x.nwp.data.float()
+
+            print(nwp_data.shape)
+            out_nwp = F.relu(self.nwp_conv0(nwp_data))
+            for i in range(0, self.number_of_conv3d_layers - 1):
+                layer = getattr(self, f"nwp_conv{i + 1}")
+                out_nwp = F.relu(layer(out_nwp))
+
+            # fully connected layers
+            out_nwp = out_nwp.reshape(batch_size, self.nwp_cnn_output_size)
+            out_nwp = F.relu(self.nwp_fc1(out_nwp))
+            out_nwp = F.relu(self.nwp_fc2(out_nwp))
+
+            # join with other FC layer
+            out = torch.cat((out, out_nwp), dim=1)
+
+        # ********************** Embedding of PV system ID ********************
+        if self.embedding_dem:
+            id = pv_system_row_number[0 : self.batch_size, 0]
+
+            id = id.type(torch.IntTensor)
+            id = id.to(out.device)
+            id_embedding = self.pv_system_id_embedding(id)
+            out = torch.cat((out, id_embedding), dim=1)
+
+        # Fully connected layers.
+        out = F.relu(self.fc3(out))
+        out = self.fc4(out)
+
+        out = out.reshape(batch_size, self.forecast_len)
+
+        return out
+
+    def load_model(
+        self,
+        local_filename: Optional[str] = "temp.ckpt",
+        remote_filename: Optional[str] = None,
+        use_hf: bool = False,
+    ):
+        """
+        Load model weights
+        """
+
+        if use_hf:
+            return Model.from_pretrained("openclimatefix/nowcasting_cnn")
+
+        if remote_filename is None:
```

Comment:
im tempted to get rid of this, mauybe keep the option to load from local_filename, but the hugging_face method should be the one to use

Reply:
Okay, yeah, sounds good!